morepath/tests/test_predicate.py

```diff
@@ -132,50 +126,48 @@ class Bar(object):
     class Other(object):
         pass
 
-    @reg.dispatch_external_predicates()
-    def f(obj, name):
-        return "fallback"
-
-    @App.predicate(f, name='model', default=None, index=ClassIndex)
+    @App.predicate(App.f, name='model', default=None, index=ClassIndex)
     def pred_obj(obj):
         return obj.__class__
 
-    @App.predicate(f, name='name', default='', index=KeyIndex, after=pred_obj)
+    @App.predicate(App.f, name='name', default='',
+                   index=KeyIndex, after=pred_obj)
     def pred_name(name):
         return name
 
-    @App.function(f, model=Foo, name='')
+    @App.function(App.f, model=Foo, name='')
     def f_foo_default(obj, name):
         return "foo default"
 
-    @App.function(f, model=Foo, name='edit')
+    @App.function(App.f, model=Foo, name='edit')
     def f_foo_edit(obj, name):
         return "foo edit"
 
-    @App.function(f, model=Bar, name='')
+    @App.function(App.f, model=Bar, name='')
     def f_bar_default(obj, name):
         return "bar default"
 
-    @App.function(f, model=Bar, name='edit')
+    @App.function(App.f, model=Bar, name='edit')
     def f_bar_edit(obj, name):
         return "bar edit"
 
     a = App()
+    a.commit()
```

Comment:
I see you have to make the same modification I had and add an explicit commit in some places. I think that's fine, but I think ``publish`` is a better place to trigger the automatic commit than ``request``.